sql/src/main/java/org/apache/druid/sql/calcite/planner/JoinHint.java

```diff
@@ -0,0 +1,81 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.sql.calcite.planner;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonValue;
+import org.apache.druid.java.util.common.IAE;
+import org.apache.druid.query.JoinAlgorithm;
+
+import java.util.Arrays;
+
+public enum JoinHint
```

Comment:
I think it would be better to not have this as an enum - and you could probably use `as(JoinAlgorithm.class)`  or `asJoinAlgorithm()`  to obtain the current value

Reply:
There's an issue with that, Calcite only seems to accept certain keywords as hints, and this is `sort_merge`. Trying to replace the hint name with `sortMerge` seems to make it not read by calcite.

Reply:
doesn't really mattter - I was just thinking something like:
```
public abstract class DruidHint {
  abstract T as(Class<T> clazz);

  public static DruidHint fromRelHint(RelHint rh) {
    // factory method
  }

  static abstrat class DruidJoinHint extends DruidHint {}

  static class SortMerge extends DruidJoinHint {
     String name() { "sort_merge"; }
     as() ... { return
  }
}
```

Reply:
Made some changes to make it more in line with this, please take a look